src/rez/developer_package.py

```diff
@@ -212,23 +228,43 @@ def _get_preprocessed(self, data):
                     )
                     return None
 
-                print_info("Applying preprocess function %s" % preprocess_func)
-
             if not preprocess_func or not isfunction(preprocess_func):
                 print_error("Function '%s' not found" % package_preprocess_function)
                 return None
 
+            return preprocess_func
+
+        with add_sys_paths(config.package_definition_build_python_paths):
+
+            preprocess_mode = PreprocessMode[self.config.package_preprocess_mode]
+            package_preprocess = _get_package_level()
+            global_preprocess = _get_global_level()
+
+            if preprocess_mode == PreprocessMode.after:
+                preprocessors = [global_preprocess, package_preprocess]
+            elif preprocess_mode == PreprocessMode.before:
+                preprocessors = [package_preprocess, global_preprocess]
+            else:
+                preprocessors = [package_preprocess or global_preprocess]
+
             preprocessed_data = deepcopy(data)
 
-            # apply preprocessing
-            try:
-                preprocess_func(this=self, data=preprocessed_data)
-            except InvalidPackageError:
-                raise
-            except Exception as e:
-                print_error("Failed to apply preprocess: %s: %s"
-                            % (e.__class__.__name__, str(e)))
-                return None
+            for preprocessor in preprocessors:
+                if not preprocessor:
+                    continue
+
+                level = "global" if preprocessor == global_preprocess else "local"
+                print_info("Applying {0} preprocess function".format(level))
```

Comment:
@nerdvegas I changed the message a little bit while fixing the messages being printed too early. I tried to make the message consistent across the local and global process. If you don't want to introduce the "global" term here, let me know and I'll try to rephrase.